### ISSUE TITLE
[bug] elective preferences are not stored

### ISSUE THREAD

**Issue by opxdelwin** [opxdelwin/plan-sync#19]
**Describe the bug**
Elective preference for year, semester and scheme isn't saved, and is disabled.

**To Reproduce**
1. Go to 'Elective Classes' screen
2. Click on select dropdown on top
3. See that 'Save Preferences' is disabled

**Expected behavior**
Preferences should be saved, switch should be active.

**Screenshots**
![image](https://github.com/opxdelwin/plan-sync/assets/84124091/ab119579-a740-4001-9457-7f257f8a0775)

**Smartphone (please complete the following information):**
 - Device: emulator
 - OS: Android 13
 - Version v0.1.23

**Additional context**
NA

**Comment by opxdelwin** [comment]
`onChanged` value should be set to 
```dart
(value) {
    setState(() {
      savePreferencesOnExit = value;
    });
},
```

**Comment by opxdelwin** [comment]
Fixed in #44
